core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -520,6 +520,8 @@ object SparkSubmit {
       }
       if (args.principal != null) {
         require(args.keytab != null, "Keytab must be specified when the keytab is specified")
+        sysProps.put("spark.yarn.keytab", args.keytab)
+        sysProps.put("spark.yarn.principal", args.principal)
```

Comment:
Hi folks, 
Yarn Client.scala checks the setting from argStrings (passed from SparkSubmit) and sparkConf (which loads java system properties starting with spark.* as well). The args settings (--principal, --keytab) will not be available in spark sql, but we can take advantage of SparkConf with system properties. Updated the pull request and tested. Please advice if this is what you are looking for, Thank you!